tour_operator/src/main/kotlin/pl/edu/pg/rsww/tour_operator/TourOperatorServer.kt

```diff
@@ -0,0 +1,27 @@
+package pl.edu.pg.rsww.touroperator
+
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.springframework.amqp.rabbit.annotation.Queue
+import org.springframework.amqp.rabbit.annotation.RabbitListener
+
+public class TourOperatorServer {
+    @RabbitListener(queuesToDeclare = [Queue("#{queueConfig.requests}")])
+    fun requestHandler(request: String): String {
+        val request = Json.decodeFromString<RequestMessage>(request)
+        if (request.path.contains("process_payment")) {
```

Comment:
idea: have some sort of payment identifier passed in the request so that we can use that in the random number generator as a seed, making it so that if an HTTP request is done multiple times, it returns the same result.

Reply:
Ok, but what problem would that solve? Since functional tests will be performed on dirty instance anyway, we would be just moving seeding portion to another service without an actual benefit for use during tests